bin/src/server/cli.rs

```diff
@@ -0,0 +1,107 @@
+use clap::{App, Arg, SubCommand};
+
+pub struct CliArgs {
+    pub port: u16,
+    pub database_url: String,
+    pub max_open_files: Option<i32>,
+    pub bulk_load_opt: Option<bool>,
+    pub sled_compression: Option<String>,
+}
+
+const PORT: &str = "PORT";
+const DATABASE_URL: &str = "DATABASE_URL";
+const ROCKSDB_MAX_OPEN_FILES: &str = "ROCKSDB_MAX_OPEN_FILES";
+const ROCKSDB_BULK_LOAD_OPTIMIZED: &str = "ROCKSDB_BULK_LOAD_OPTIMIZED";
+const SLED_COMPRESSION: &str = "SLED_COMPRESSION";
```

Comment:
No need to separate this out to constants, since none of the other string literals are constants either

Reply:
Actually the reason I extracted this out to constants was because the name of the arg was reused during creation and matching. 
e.g
```rust
let database_path_argument = Arg::with_name(DATABASE_PATH)
...
```
and then the argument is matched and extracted read via
```rust
path: String::from(matches.value_of(DATABASE_PATH).unwrap()),
```

Reply:
Got it, that makes sense